src/hooks.tsx

```diff
@@ -0,0 +1,16 @@
+import { useQuery } from "@tanstack/react-query";
+import { getRestaurantList } from "./api";
+import { fromJsonToProp } from "./utils";
+
+function useFetchQuery(range: number) {
+  const { isLoading, isError, data } = useQuery({
+    queryKey: ["retrieve-list", range],
+    queryFn: async () => {
+      const prom: JSON = await getRestaurantList(range);
+      return fromJsonToProp(prom);
+    },
+  });
+  return { data, isLoading, isError };
+}
```

Comment:
A couple of minor suggestions:
* consider using a more explicit name for the hook, so that when you have multiples they are easily distinguishable. For example, it could be based on the name of the fetching function: `getRestaurantList -> useGetRestaurantList`.
* instead of extracting the specific fields you need now, consider returning the whole result of `useQuery`:
```suggestion
function useFetchQuery(range: number) {
  return useQuery({
    queryKey: ["retrieve-list", range],
    queryFn: async () => {
      const prom: JSON = await getRestaurantList(range);
      return fromJsonToProp(prom);
    },
  });
}
```